uvicorn/server.py

```diff
@@ -252,7 +252,7 @@ async def on_tick(self, counter: int) -> bool:
             return True
         if self.config.limit_max_requests is not None:
             return self.server_state.total_requests >= self.config.limit_max_requests
-        return False
+        return self.lifespan.should_exit
```

Comment:
```suggestion
        # Determine if we should exit.
        if self.should_exit or self.lifespan.should_exit:
            return True
        if self.config.limit_max_requests is not None:
            return self.server_state.total_requests >= self.config.limit_max_requests
        return False
```

would that suggestion be marginally better ? ie we test for exit in `on_tick` on both the server `should_exit` flag and the lifespan one instead of delaying it at the end ? either way is ok to me though